design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,101 @@
+  - `rook-num-csi-rbd-plugin-pods`
+  - `rook-num-csi-cephfs-plugin-pods`
+  - `rook-num-csi-nfs-plugin-pods`
+- `rook-num-nodes` - Number of nodes on which the cluster has running pods.
+  - Since clusters with portable PVCs have one "node" per PVC, this will help show the actual node
+    count for Rook installs in portable environments
+  - We can get this by counting the number of crash collector pods; however, if any users disable
+    the crash collector, we should report an unknown value, possibly `unknown`.
+- `rook-num-storage-classes` - Number of storage classes related to the current Ceph cluster. We can
+  also break this up by specific kinds of storage classes, but the overall may be useful if new
+  classes are added in the future.
+  - `rook-num-rbd-storage-classes`
+  - `rook-num-cephfs-storage-classes`
+  - `rook-num-nfs-storage-classes`
+  - `rook-num-bucket-storage-classes`
+- `rook-num-storage-class-device-sets` - Number of storage class device sets.
+- `rook-num-portable-storage-class-device-sets` - .
+- `rook-mons-are-pvc` - true/false whether mons are on PVC.
+- `rook-network-provider` - The network provider used for the cluster. e.g., default, host, multus.
+- `rook-external-mode` - true/false if the cluster is in external mode.
+
+## Information Rook is interested in that is already collected by Ceph telemetry
+- RBD pools (name is stripped), some config info, and the number of them.
+- Ceph Filesystems and MDS info.
+- RGW count, zonegroups, zones.
+- RBD mirroring info.
+
+## Metrics that Rook would like to see added to Ceph's telemetry
+- Number of NFS clusters
+  - Ceph can count NFS clusters by listing the namespaces in the `.nfs` pool and report those
+    namespaces that have `conf-nfs.<namespace>` objects present
+- Filesystem mirroring info
+- RGW bucket notification usage (Yuval may have good detail on this point)
+
+## Proposed integration strategy
+- Rook will first merge [`rook-version`](https://github.com/rook/rook/pull/10161).
+- Rook, with input from Ceph telemetry, will approve a version of this design doc and treat all
+  noted metric names as requirements.
+- Ceph will add all of the noted telemetry keys from the config-key database to its telemetry and
+  backport to Ceph Pacific and Ceph Quincy.
+  - Ceph code should handle the case where Rook has not set these fields.
+  - Ceph code can add `rook-version` immediately if they expect it might get into the next
+    Pacific/Quincy release more quickly.
+- Rook will implement each metric (or related metric group) individually over time.
+
+This strategy will allow Rook time to add telemetry items as it is able without rushing. Because the
+telemetry fields will be approved all at once, it will also minimize the coordination that is
+required between Ceph and Rook. The Ceph team will not need to create PRs one-to-one with Rook, and
+we can limit version mismatch issues as the telemetry is being added.
+
+## Rejected metrics
```

Comment:
PR comments could keep track of rejected metrics. Why track them in the design doc?

Reply:
While that is true, PR comments are often hidden, especially for PRs with lots of comments. This way, the distillation can be seen by anyone looking at the doc without having to find the PR and then un-hide all the comments from who knows how long ago. In my experience, a good design doc must mention discarded ideas so that people are less likely to suggest and later implement ideas that should be rejected once we all have forgotten what was rejected and why.